shared-module/displayio/__init__.c

```diff
@@ -1,91 +1,188 @@
-#include "shared-bindings/displayio/FourWire.h"
 
-extern displayio_fourwire_obj_t board_display_obj;
+#include <string.h>
+#include "shared-module/displayio/__init__.h"
 
-void start_region_update(displayio_fourwire_obj_t* display, uint16_t x0, uint16_t y0, uint16_t x1, uint16_t y1) {
-    // TODO delegate between different display types
-    displayio_fourwire_start_region_update(display, x0, y0, x1, y1);
-}
+#include "shared-bindings/displayio/Bitmap.h"
+#include "shared-bindings/displayio/Display.h"
+#include "shared-bindings/displayio/Group.h"
+#include "shared-bindings/displayio/Palette.h"
+#include "shared-bindings/displayio/Sprite.h"
+#include "supervisor/usb.h"
 
-void finish_region_update(displayio_fourwire_obj_t* display) {
-    // TODO delegate between different display types
-    displayio_fourwire_finish_region_update(display);
-}
+primary_display_t displays[CIRCUITPY_DISPLAY_LIMIT];
 
-void finish_refresh(displayio_fourwire_obj_t* display) {
-    // TODO delegate between different display types
-    displayio_fourwire_finish_refresh(display);
-}
+void displayio_refresh_displays(void) {
+    for (uint8_t i = 0; i < CIRCUITPY_DISPLAY_LIMIT; i++) {
+        if (displays[i].display.base.type == NULL || displays[i].display.base.type == &mp_type_NoneType) {
+            continue;
+        }
+        displayio_display_obj_t* display = &displays[i].display;
 
-bool frame_queued(displayio_fourwire_obj_t* display) {
-    // TODO delegate between different display types
-    return displayio_fourwire_frame_queued(display);
-}
+        if (!displayio_display_frame_queued(display)) {
+            return;
+        }
+        if (displayio_display_refresh_queued(display)) {
+            // We compute the pixels
+            uint16_t x0 = 0;
+            uint16_t y0 = 0;
+            uint16_t x1 = display->width;
+            uint16_t y1 = display->height;
+            size_t index = 0;
+            //size_t row_size = (x1 - x0);
+            uint16_t buffer_size = 256;
```

Comment:
Why is `buffer_size` a variable instead of a #define or a `const`?

Reply:
No real reason. I'm likely to change this all soon to make it faster so I'd prefer to just leave it as is.

Reply:
kk